src/devices/Bmxx80/Bme680.cs

```diff
@@ -0,0 +1,230 @@
+        /// Default I2C bus address.
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x76;
+
+        /// <summary>
+        /// Secondary I2C bus address.
+        /// </summary>
+        public const byte SecondaryI2cAddress = 0x77;
+
+        /// <summary>
+        /// The expected chip ID of the BME680.
+        /// </summary>
+        private const byte DeviceId = 0x61;
+
+        /// <summary>
+        /// Calibration data for the <see cref="Bme680"/>.
+        /// </summary>
+        private readonly Bme680CalibrationData _bme680Calibration;
+
+        /// <summary>
+        /// Initialize a new instance of the <see cref="Bme680"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Bme680(I2cDevice i2cDevice)
+            : base(DeviceId, i2cDevice)
+        {
+            var bme680CalibrationData = new Bme680CalibrationData();
+            bme680CalibrationData.ReadFromDevice(this);
+            _bme680Calibration = bme680CalibrationData;
+            _calibrationData = bme680CalibrationData;
+
+            _communicationProtocol = CommunicationProtocol.I2c;
+            _controlRegister = (byte)Bme680Register.CTRL_MEAS;
+        }
+
+        /// <summary>
+        /// Set the humidity sampling.
+        /// </summary>
+        /// <param name="sampling">The <see cref="Sampling"/> to set.</param>
+        public void SetHumiditySampling(Sampling sampling)
+        {
+            var register = (byte)Bme680Register.CTRL_HUM;
+            byte read = Read8BitsFromRegister(register);
+
+            // Clear first 3 bits.
+            var cleared = (byte)(read & 0b_1111_1000);
+
+            _i2cDevice.Write(new[] { register, (byte)(cleared | (byte)sampling) });
+        }
+
+        /// <summary>
+        /// Sets the power mode to the given mode
+        /// </summary>
+        /// <param name="powerMode">The <see cref="Bme680PowerMode"/> to set.</param>
+        public void SetPowerMode(Bme680PowerMode powerMode)
+        {
+            byte read = Read8BitsFromRegister(_controlRegister);
+
+            // Clear first 2 bits.
+            var cleared = (byte)(read & 0b_1111_1100);
+
+            _i2cDevice.Write(new[] { _controlRegister, (byte)(cleared | (byte)powerMode) });
+        }
+
+        /// <summary>
+        /// Read a value indicating whether or not new sensor data is available.
+        /// </summary>
+        /// <returns>True if new data is available.</returns>
+        public bool ReadHasNewData()
+        {
+            byte read = Read8BitsFromRegister((byte)Bme680Register.STATUS);
+
+            // Get only the power mode bit.
+            var hasNewData = (byte)(read & 0b_1000_0000);
+
+            return (hasNewData >> 7) == 1;
+        }
+
+        /// <summary>
+        /// Read the humidity.
+        /// </summary>
+        /// <returns>Calculated humidity.</returns>
+        public double ReadHumidity()
```

Comment:
For consistency sake and given that we also support PowerMode with the 680, should we make these methods Async and return a Task<double> which does something like:

```c#
            if (ReadPowerMode() == Bmx280PowerMode.Forced)
            {
                await Task.Delay(GetMeasurementTimeForForcedMode(ReadHumiditySampling()));
            }
```

Reply:
Agree with making it consistent but the BME680 doesn't have any specified wait times in the datasheet for reading data like the BMx280 sensors do. Instead, there's a flag that indicates when new data is available to be read. Thoughts?

Reply:
If we really wanted to follow the framework design guidelines and designed this like a corefx class in here, we would have something like this:

```c#
public abstract class SensorBaseClass
{
    public double ReadHumidity()
    {
        // Perform Setup, validation and other things that all devices have in common.
        return ReadHumidityCore();
    }

    public Task<double> ReadHumidityAsync()
    {
        //Perform Setup, validation and other things that all devices have in common.
        return ReadHumidityAsyncCore();
    }

    protected abstract double ReadHumidityCore();
    protected virtual Task<double> ReadHumidityAsyncCore()
    {
        return Task.FromResult(ReadHumidityCore());
    }
}

public class MySensorWhichDoesntWorkAsync : SensorBaseClass
{
    protected override double ReadHumidityCore()
    {
        // perform sync read here
    }
}

public class MySensorWhichWorksAsync : SensorBaseClass
{
    protected override Task<double> ReadHumidityAsyncCore()
    {
        // perofrm async read.
    }
    protected override double ReadHumidityCore()
    {
        return ReadHumidityAsyncCore().GetAwaiter().GetResult();
    }
}
```

All of that said, you don't have to do this now in this PR, for now, what you can do is to simply also offer a `ReadHumidityAsync` method (and all other values) which simply return a `Task.FromResult(ReadHumidity())`. I guess you get to choose if you want to do the full design patter from the framework guidelines, or if you just want to add consistency for now.

Reply:
I think ultimately the corefx class pattern is the way to go... However, I've gone with consistency _for now_ as there's quite a lot in this PR and it would be great to get this in 😄